x11-plugins/pidgin-telegram/pidgin-telegram-1.3.1.ebuild

```diff
@@ -0,0 +1,57 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+DESCRIPTION="A libpurple protocol plugin that adds support for the Telegram messenger"
+HOMEPAGE="https://github.com/majn/telegram-purple"
+SRC_URI="https://github.com/majn/telegram-purple/releases/download/v${PV}/telegram-purple_${PV}.orig.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+KEYWORDS="~amd64 ~x86"
+SLOT="0"
+IUSE="gcrypt +icons +nls +webp +zlib"
+
+RDEPEND="net-im/pidgin
+	gcrypt? ( dev-libs/libgcrypt:0= )
+	!gcrypt? ( dev-libs/openssl:0= )
+	nls? ( sys-devel/gettext )
+	webp? ( media-libs/libwebp )
+	zlib? ( sys-libs/zlib )"
+
+DEPEND="${RDEPEND}"
+
+S="${WORKDIR}/telegram-purple"
+
+DOCS=( "AUTHORS" "CHANGELOG.md" "HACKING.md" "HACKING.BUILD.md" "README.md" )
+
+src_prepare() {
+	default
+
+	# Remove '-Werror' to make it compile
+	local werror_file
+	local WERROR_FILES=( "Makefile.mingw" "Makefile.tgl.mingw" "tgl/Makefile.in" "tgl/tl-parser/Makefile.in" )
+	for werror_file in "${WERROR_FILES[@]}"; do
+		sed -i -e 's/-Werror //' "${werror_file}" || die
+	done
+}
+
+src_configure() {
+	local myeconfargs=(
+		$(use_enable gcrypt)
+		$(use_enable nls translation)
+		$(use_enable webp libwebp)
+		$(use_with zlib)
+	)
+
+	econf "${myeconfargs[@]}"
+}
+
+src_install() {
+	if ! use icons; then
+		emake DESTDIR="${D}" noicon_install
```

Comment:
I dare say moving the conditional into `$(usex icons ...)` would be cleaner than conditionally using default.

Reply:
Since I removed noicon_install, this won't be needed anymore.